lib/src/util.dart

```diff
@@ -15,7 +17,8 @@ String buildQueryString(Map<String, dynamic> params) {
     if (params[key] == null) {
       continue;
     }
-    queryString.write('$key=${Uri.encodeComponent(params[key].toString())}');
+    // TODO: change to library-specific json encoding
+    queryString.write('$key=${Uri.encodeComponent(jsonEncode(params[key]))}');
```

Comment:
Can you explain why we're `jsonEncode`ing all objects that are values? This makes me a little nervous that we might break other things so I'd like to understand it better? 🤷 

> change to library-specific json encoding
Also, I'm not sure I know what you mean by that. Could you give me more information so when I or another person wants to address this TODO we know what we're doing?

Reply:
You're right. I made this change following the edit of the type of accepted values in this `params` map (`String` to `dynamic`), but on second thought it's a little risky.

This TODO was related to my other PR, I'll remove it.

My idea was to call an encoding function to serialize values that wouldn't be strings.